### PR TITLE
[python] add mediatype to setinfo for music

### PR DESCRIPTION
**Pull request by BigNoid**
Enables scripts to set dbtype so context/info will work for listproviders.
This is just a c/p from the video mediatype plus moving m_type from protected to public

### CODE REVIEW

File: xbmc/interfaces/legacy/ListItem.h [hunk 1]
```diff
@@ -240,6 +240,7 @@ namespace XBMCAddon
        *     - lyrics        : string (On a dark desert highway...)
        *     - playcount     : integer (2) - number of times this item has been played
        *     - lastplayed    : string (%Y-%m-%d %h:%m:%s = 2009-04-05 23:16:04)
+       *     - mediatype     : string - "song", "album", "artist"
```
Comment (phil65) [inline xbmc/interfaces/legacy/ListItem.h]: according to IsValidMediaType(), "music" is also valid. (I have no idea what it should be used for though)

**Comment by phil65** [comment]
@tamland 

@BigNoid perhaps you could also cherrypick https://github.com/Montellese/xbmc/commit/f5a0fe404d52132311149d0e8bf41c0184ab5438https://github.com/Montellese/xbmc/commit/f5a0fe404d52132311149d0e8bf41c0184ab5438 from @Montellese ?

**Comment by DaveTBlake** [comment]
For consistency with the rest of CMusicInfoTag I think this should add a SetType method (GetType already exists) rather than make m_type variable public.

**Comment by Montellese** [comment]
Yes please don't expose members directly. Implement a setter instead.

**Comment by BigNoid** [comment]
Updated with SetType method.

@phil65 It would need some work in PictureInfoTag as well I think, I couldn't quite figure out what is needed exactly there in any case, so maybe thats something for another PR.

**Comment by phil65** [comment]
ok, sure. Thx.

**Comment by MartijnKaijser** [comment]
jenkins build this please
Good for merge?

**Pull request merged**